Taskfile.yml

```
@@ -40,6 +40,66 @@ tasks:
     cmds:
       - pushd mkdocs && uv run mkdocs serve
 
+  ##
+  ## Testing and Linting
+  ##
+  test:
+    desc: Run all documentation tests and linting checks
+    deps:
+      - deps/patch
+    cmds:
+      - task: lint
+      - echo "All documentation tests passed!"
+
+  lint:
+    desc: Run all linting checks (spelling, markdown)
+    deps:
+      - deps/patch
+    cmds:
+      - task: lint:spelling
+      - task: lint:markdown
+
+  lint:spelling:
+    desc: Check spelling in documentation
+    deps:
+      - deps/patch
+    dir: mkdocs
+    cmds:
+      - uv run codespell --config ../.codespellrc ../docs
+
+  lint:markdown:
+    desc: Check Markdown syntax and style
+    deps:
+      - deps/patch
+    dir: mkdocs
+    cmds:
+      - uv run pymarkdown --config ../.pymarkdown scan ../docs
+
+  lint:fix:
+    desc: Auto-fix spelling and markdown issues where possible
+    deps:
+      - deps/patch
+    dir: mkdocs
+    cmds:
+      - uv run codespell --config ../.codespellrc ../docs --write-changes
+      - uv run pymarkdown --config ../.pymarkdown fix -r ../docs
+
+  lint:fix:spelling:
+    desc: Auto-fix spelling issues only
+    deps:
+      - deps/patch
+    dir: mkdocs
+    cmds:
+      - uv run codespell --config ../.codespellrc ../docs --write-changes
+
+  lint:fix:markdown:
+    desc: Auto-fix markdown issues only
+    deps:
+      - deps/patch
+    dir: mkdocs
+    cmds:
+      - uv run pymarkdown --config ../.pymarkdown fix -r ../docs
```

```suggestion
  lint:fix:
    desc: Auto-fix spelling and markdown issues where possible
    deps:
      - deps/patch
    dir: mkdocs
    cmds:
      - task: lint:fix:spelling
      - task: lint:fix:markdown

  lint:fix:spelling:
    desc: Auto-fix spelling issues only
    internal: true
    deps:
      - deps/patch
    dir: mkdocs
    cmds:
      - uv run codespell --config ../.codespellrc ../docs --write-changes

  lint:fix:markdown:
    desc: Auto-fix markdown issues only
    internal: true
    deps:
      - deps/patch
    dir: mkdocs
    cmds:
      - uv run pymarkdown --config ../.pymarkdown fix -r ../docs
```

Same comment here regarding internal tasks.
Lets reuse the tasks in the main task rather than redefining.